WordPressKit/IPLocationRemote.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+/// Remote type to fetch the user's IP Location using the public `geo` API.
+///
+public final class IPLocationRemote {
+    private enum Constants {
+        static let jsonDecoder = JSONDecoder()
+    }
+
+    private let urlSession: URLSession
+
+    init(urlSession: URLSession = URLSession.shared) {
+        self.urlSession = urlSession
+    }
+
+    /// Fetches the country code from the device ip.
+    ///
+    public func fetchIPCountryCode(completion: @escaping (Result<String, Error>) -> Void) {
+        let path = WordPressComOAuthClient.WordPressComOAuthDefaultApiBaseUrl + "/geo/"
+        guard let url = URL(string: path) else {
+            return completion(.failure(IPLocationError.malformedURL))
+        }
+
+        let request = URLRequest(url: url)
+        let task = urlSession.dataTask(with: request) { data, _, error in
+            guard let data else {
+                return
+            }
```

Comment:
It would be great **not** to handle this silently. We can do the following:
```swift
let data = data ?? Date()
```
Then the decoding is going to fail.